classes/submissionFile/Repository.inc.php

```diff
@@ -722,7 +722,7 @@ public function getWorkflowStageId(SubmissionFile $submissionFile): ?int
             $query = $queryDao->getById($note->getAssocId());
 
             // The note should be associated with a query. If not, fail.
-            if ($query->getAssocType() != PKPApplication::ASSOC_TYPE_QUERY) {
+            if (is_null($query) || ($query->getAssocType() != PKPApplication::ASSOC_TYPE_QUERY)) {
```

Comment:
As we've bumped the PHP to v8, it's possible to use this alternative syntax: `$query?->getAssocType() !== PKPApplication::ASSOC_TYPE_QUERY`.

Reply:
Nice one. Changed.